velox/exec/RowContainer.h

```diff
@@ -238,7 +238,7 @@ class RowContainer {
       int32_t numRows,
       RowColumn col,
       vector_size_t resultOffset,
-      VectorPtr& result);
+      const VectorPtr& result);
```

Comment:
How hard is it to change to `BaseVector&` here?

Reply:
Hi Jimmy, BaseVector could be sufficient here. I presumed we just wanted to keep it safe to ensure that RowContainer always had a valid vector to copy to in these APIs.

@mbasmanova : Is there any reason for shared_ptr usage ?

Reply:
BaseVector would be nicer.

Reply:
We need this PR for some changes in WindowFunction use. But @pramodsatya can try what it takes for using BaseVector& in a subsequent PR.